projectq/ops/_gates.py

```diff
@@ -338,3 +338,58 @@ def get_inverse(self):
 
 #: Shortcut (instance of) :class:`projectq.ops.BarrierGate`
 Barrier = BarrierGate()
+
+
+class FlipBits(SelfInverseGate):
+    """ Gate for transforming qubits in state |0> to any computational basis state """
+    def __init__(self, bits_to_flip):
+        """
+        Initialize FlipBits gate.
+
+        Example:
+            .. code-block:: python
+
+                qureg = eng.allocate_qureg(2)
+                FlipBits([0, 1]) | qureg
+
+        Note:
+            The amplitude of state k is final_state[k]. When the state k is
+            written in binary notation, then qureg[0] denotes the qubit
+            whose state corresponds to the least significant bit of k.
+
+        Args:
+            bits_to_flip(list[int]|list[bool]|str|int): int or array of 0/1, True/False, or
+                                    string of 0/1 identifying
+                                    the qubits to flip. In case of int, the bits to flip are
+                                    determined from the binary digits, with the least significant
+                                    bit corresponding to qureg[0]. If bits_to_flip is negative,
+                                    exactly all qubits which would not be flipped for the input
+                                    -bits_to_flip-1 are flipped, i.e., bits_to_flip=-1 flips all
+                                    qubits.
+        """
+        if isinstance(bits_to_flip, int):
+            self.bits_to_flip = bits_to_flip
+        else:
+            self.bits_to_flip = 0
+            for c in reversed(list(bits_to_flip)):
+                bit = 0b1 if c == '1' or c == 1 or c == True else 0b0
+                self.bits_to_flip = (self.bits_to_flip << 1) | bit
+
+    def __str__(self):
+        return "FlipBits("+str(self.bits_to_flip)+")"
+
+    def __or__(self, qubits):
+        if isinstance(qubits, tuple) and len(qubits) > 1:
```

Comment:
A simple solution would be
```
quregs_tuple = self.make_tuple_of_qureg(qubits)
if len(quregs_tuple) > 1:
...
```

Reply:
Yes, marginally simpler. Good suggestion.